src/package.h

```diff
@@ -16,6 +16,45 @@ class Package {
   public:
     typedef boost::shared_ptr<Package> Ptr;
 
+    /// Returns optimal fill mass for a resource to be packaged. Can be used
+    /// to determine how to respond to requests for material, and to actually
+    /// package and send off trades.
+    /// Packaging strategy "first" simply fills the packages one by one to the
+    /// maximum fill. Therefore, it should always try to max fill.
+    /// Packaging strategy "equal" tries to fill all packages to the same mass.
+    /// This tries to find the optimal number and fill mass of packages given
+    /// the packaging limitations. It does this by calculating bounding fills, 
+    /// floor(quantity/fill_min) and ceiling(quantity/fill_max). 
+    /// There might be a scenario where there is no solution, i.e. an integer
+    /// number of packages cannot be filled with no remainder. In this case,
+    /// the most effective fill strategy is to fill to the max. Numeric example:
+    /// quantity = 5, fill_min = 3, fill_max = 4. num_min_fill = floor(5/3) = 1,
+    /// num_max_fill = ceil(5/4) = 2. num_min_fill < num_max_fill, so fill to
+    /// the max.
+    double GetFillMass(Resource::Ptr r);
+
+    /// Repackages a single resource into a package. If some quantity of the 
+    /// resource cannot be packaged using the given packaging strategy and
+    /// restrictions, the remainder is left in the resource object. 
+    template <class T>
+    std::vector<typename T::Ptr> PackageResource(typename T::Ptr r) {
```

Comment:
Probably best to move this to the CC file instead of inline here.

Reply:
Also- do you use this anywhere yet?

Reply:
No